packages/ef-runtime-client/src/EFRuntime/EFRuntime.ts

```diff
@@ -0,0 +1,87 @@
+import { IComponentRegistry } from "../ComponentRegistry";
+import { ModuleLoader } from "../ModuleLoader";
+import { StylingHandler } from "../StylingHandler";
+
+export interface IRuntimeDependencies {
+  componentRegistry: IComponentRegistry;
+  moduleLoader: ModuleLoader;
+  stylingHandler: StylingHandler;
+}
+
+export class EFRuntime {
+  private registry: IComponentRegistry;
+  private moduleLoader: ModuleLoader;
+  private stylingHandler: StylingHandler;
+
+  constructor({
+    componentRegistry,
+    moduleLoader,
+    stylingHandler,
+  }: IRuntimeDependencies) {
+    this.registry = componentRegistry;
+    this.moduleLoader = moduleLoader;
+    this.stylingHandler = stylingHandler;
+  }
+
+  private validateOptions(options: {
+    systemCode?: string;
+    overrides?: { [propName: string]: string };
+  }): void {
+    if (!options.systemCode) {
+      throw new Error("Must provide a systemCode option");
+    }
+  }
+
+  async init(
+    options: {
+      systemCode?: string;
+      overrides?: { [propName: string]: string };
+    } = {}
+  ): Promise<void> {
+    this.validateOptions(options);
+
+    await Promise.all([
+      this.moduleLoader.init(),
+      this.registry.fetch(options.systemCode as string),
+    ]);
+
+    if (options.overrides) {
+      this.registry.applyOverrides(options.overrides);
+    }
+
+    this.loadAll();
+  }
+
+  loadAll(): void {
+    for (const component of this.registry.getComponentKeys()) {
+      this.load(component).catch((error) =>
+        console.error(`Error loading ${component}`, error)
+      );
+    }
+  }
+
+  async load(component: string): Promise<void> {
+    const url = this.registry.getURL(component);
+    if (!url) {
+      console.error(
+        `Component ${component} was not found in the Component Registry`
+      );
+      return;
+    }
+    this.stylingHandler.addStyling(url);
+    try {
+      const componentModule = await this.moduleLoader.importModule(`${url}/js`);
```

Comment:
This and addStyling are a bit inconsistent - one adds JS before call, and the other adds CSS after call.

Maybe they should _only_ take URL, and the registry definition should include both?

e.g.
`{ jsUrl: "example.org/blah", cssUrl: "example.org/blah" }`

Reply:
Yea I think this is a great point we have a ticket to make sure we deal with [this](https://financialtimes.atlassian.net/browse/EF-37?atlOrigin=eyJpIjoiMWNjNmMzZWRlNmY5NDJmMmJiMGI4N2U4ODJiYjIzZTgiLCJwIjoiaiJ9)!